packages/codemirror-lang-dentaku/src/linting.ts

```diff
@@ -18,29 +19,45 @@ export interface ErrorMessages {
   undefinedVariable: string;
   /** Missing child property. */
   missingChildProperty: string;
-  /** Expected at least one parameter in a function call. */
-  expectedAtLeastOneParameter: string;
-  /** Expected that many parameters, found this many. */
-  parameterCountMismatch: (
-    expectedAmount: number,
-    actualAmount: number
+  /** Expected exactly that many parameters, found this many. */
+  expectedExactArgumentCount: (count: number) => string;
+  /** Expected at least that many parameters, found this many. */
+  expectedMinimumArgumentCount: (count: number) => string;
+  /** Expected between that and that many parameters, found this many. */
+  expectedArgumentCountRange: (
+    actualCount: number,
+    minCount: number,
+    maxCount: number
   ) => string;
 }
 
 export interface DentakuLinterOptions {
   /** List of variable names to not treat as undefined. */
   knownVariables?: Array<string>;
+  /** Custom functions. */
+  customFunctions?: Record<string, DentakuFunctionConfig>;
   /** Custom error messages. */
   messages?: ErrorMessages;
 }
 
 export function dentakuLinter({
   knownVariables = [],
+  customFunctions = {},
   messages = defaultErrorMessages,
 }: DentakuLinterOptions = {}): LintSource {
   return (view) => {
     const diagnostics: Diagnostic[] = [];
 
+    const knownFunctions: Record<string, DentakuFunctionConfig> = {
+      ...(builtInFunctions as unknown as Record<
+        BuiltInFunctionsType,
+        DentakuFunctionConfig
+      >),
```

Comment:
question: why is this cast necessary?

Reply:
It is not. It is also a remnant from the testings with reusable config for `linting` and `autocompletion`. I'll remove it.